src/main.h

```diff
@@ -426,6 +429,7 @@ class CScriptCheck
         std::swap(nFlags, check.nFlags);
         std::swap(cacheStore, check.cacheStore);
         std::swap(error, check.error);
+        std::swap(cachedHashesMap, check.cachedHashesMap);
```

Comment:
My logic was based on the presence of the lock implies multi-threaded access,  and this operator implies the entire memory is being swapped when `swap()` is called. 
To me,  that would imply that it should be `LOCK`'d to avoid another thread reading it during this swap.